### ISSUE TITLE
No Label for Search Block Dropdown Form element

### ISSUE THREAD

**Issue by ThomasKindermann** [plone/volto#5400]
**Describe the bug**
The "sort by" dropdown in the Search block doesn't have a label. 
It should have a label for A11y.

**To Reproduce**

1. Go to any volto page
2. Create a Page with the Search block.
3. Use an A11y Tool or inspect the Dropdown
4. Dropdown / Input Form element does not have a label.

**Expected behavior**
Any Form Elements should have a label for screen readers.

**Screenshots**
![Screenshot 2023-11-14 at 12 01 05](https://github.com/plone/volto/assets/50659809/61ab209b-3601-4aba-b7ff-6a7f6fb30bcf)

- OS: OSX
- Browser Chrome
- Volto 17.0.0-alpha.24
- Plone 6.0.6
- plone.restapi 8.41.0

**Comment by Nandini99-git** [comment]
Hi @ThomasKindermann, I would like to work on this. Please assign me

**Comment by pranayjoshi** [comment]
Hi @ThomasKindermann , I am working on this. Is 'Sort by Dropdown' fine as a label? or shall I use the dropdown value as the label?

**Comment by stevepiercy** [comment]
In English the phrase preceding the sort input should be `Sort by`, not `Sort on`. I hate to pile on to an issue, but since it is part of the same block, it may as well be corrected at the same time. I do not suggest refactoring the code to replace `sortOn*` names, as I don't know what impact that would have and going that far might be a breaking change.

For the sort input, I would suggest the label `Sort by selected option`. I thought about just `selected option`, but I doubt that screen readers would be able to connect the text before the input to the actual input.

The label for the sort input must be internationalized. See https://6.docs.plone.org/volto/recipes/i18n.html.

**Comment by shivaansh0610-LUFFY** [comment]
Hi, @stevepiercy 
Is this issue open?
If it is open, I will be happy to work on it.

**Comment by stevepiercy** [comment]
@shivaansh0610-LUFFY using your super power of observation, you may see that the green "Open" icon reflects its status.

**Comment by shivaansh0610-LUFFY** [comment]
Thanks, I’ll take a look and start working on this.
